ui/src/map/layers/switch/switch-layer.ts

```diff
@@ -31,9 +31,11 @@ export function createSwitchLayer(
 ): MapLayer {
     const layerId = ++newestLayerId;
     const getSwitchesFromApi = () => {
-        return Promise.all(
-            mapTiles.map((t) => getSwitchesByTile(changeTimes.layoutSwitch, t, publishType)),
-        ).then((switchGroups) => switchGroups.flat().filter(filterUniqueById((s) => s.id)));
+        return resolution <= Limits.SWITCH_SHOW
+            ? Promise.all(
+                  mapTiles.map((t) => getSwitchesByTile(changeTimes.layoutSwitch, t, publishType)),
+              ).then((switchGroups) => switchGroups.flat().filter(filterUniqueById((s) => s.id)))
+            : getSwitches(selection.selectedItems.switches, publishType);
```

Comment:
Le biff tällä puolella. Jos ollaan zoomattu ulos vaihteiden näyttämisen resoluutiosta, niin halutut vaihteet haetaan vaan yksittäin ei-tiilitetyllä haulla.